### PR TITLE
Updated with ORM

### PR DESCRIPTION
**Pull request by szhansmoothstack**
Should work with ORM pulled from master

### CODE REVIEW

File: order-api/src/main/java/com/smoothstack/order/DemMain.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.autoconfigure.security.servlet.SecurityAutoConfiguration;
+
+//@SpringBootApplication (exclude = {SecurityAutoConfiguration.class})
+public class DemMain {
+//    public static void main(String[] args) {
+//        SpringApplication.run(DemoApplication.class, args);
+//    }
+}
```
Comment (ezramitchell) [inline order-api/src/main/java/com/smoothstack/order/DemMain.java]: What is this file for?

File: order-api/src/main/java/com/smoothstack/order/Main.java [hunk 2]
```diff
@@ -51,5 +53,4 @@ public int getExitCode() {
         }
 
     }
-
 }
```
Comment (ezramitchell) [inline order-api/src/main/java/com/smoothstack/order/Main.java:56]: CORS mappings belong in the Shared security module. It's already there actually.

File: order-api/src/main/java/com/smoothstack/order/api/OrderApi.java [hunk 3]
```diff
@@ -41,20 +42,18 @@
...
         @ApiResponse(code = 401, message = "Access token is missing or invalid", response = String.class),
         @ApiResponse(code = 403, message = "Forbidden") })
     @PutMapping(
-        value = "/order",
+        path = "/order",
         produces = { "application/json", "application/xml" },
         consumes = { "application/json", "application/xml" }
     )
-    default ResponseEntity<InlineResponse200> createOrder(@ApiParam(value = ""  )  @Valid @RequestBody(required = false) Order order) {
+    default ResponseEntity<?> createOrder(@ApiParam(value = ""  )  @Valid @RequestBody(required = false) Order order) {
```
Comment (ezramitchell) [inline order-api/src/main/java/com/smoothstack/order/api/OrderApi.java]: Don't return generics, throw exceptions and use controller advice or @exceptionhandlers

File: order-api/src/main/java/com/smoothstack/order/api/OrderApiController.java [hunk 4]
```diff
@@ -1,24 +1,63 @@
...
     }
 
+    @Override
+    @PreAuthorize("hasRole('ROLE_user')")
+    public ResponseEntity<List<Order>> getOrder(@ApiParam(value = "if true only returns pending orders") @Valid @RequestParam(value = "active", required = false) Boolean active){
+        return orderService.getOrder(true);
+    }
+
+    @Override
+    public ResponseEntity<?> createOrder(Order order) {
```
Comment (ezramitchell) [inline order-api/src/main/java/com/smoothstack/order/api/OrderApiController.java]: Same here, use exception handlers not generics

File: order-api/src/main/java/com/smoothstack/order/service/OrderService.java [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+                orderRepo.findById(id).get() : null;
+        if (orderEntity == null) return;
+        List <FoodOrderEntity> foodOrderEntityList = orderEntity.getItems();
+        for (FoodOrderEntity foodOrderEntity : foodOrderEntityList)
+            foodOrderRepo.delete(foodOrderEntity);
+        orderRepo.deleteById(id);
+    }
+
+    @Override
+    public ResponseEntity<?> createOrder(@ApiParam(value = "")
```
Comment (ezramitchell) [inline order-api/src/main/java/com/smoothstack/order/service/OrderService.java]: # No generics

**Review by ezramitchell — CHANGES_REQUESTED** [review]
Only problem is returning generics, look into controller advice and exception handlers instead.

File: order-api/src/main/java/com/smoothstack/order/DemMain.java (hunk 1 above)
Comment (szhansmoothstack) [reply order-api/src/main/java/com/smoothstack/order/DemMain.java]: literally nothing, just delete

**Review by ezramitchell — APPROVED** [review]

**Pull request merged**